v0/util/index.js

```diff
@@ -1474,6 +1479,110 @@ function getValidDynamicFormConfig(
   return res;
 }
 
+/**
+ * error handler during transformation of a single rudder event for rt transform destinaiton
+ *
+ * This function is to be used only when we have a simple error handling scenario
+ * If some customisation wrt error handling is required, we recommend to not use this function
+ *
+ * @param {object} input - single rudder event
+ * @param {*} error - error occurred during transformation of a single rudder event
+ * @param {string} destType - destination name
+ * @returns
+ */
+const handleRtTfSingleEventError = (input, error, destType) => {
+  const errObj = generateErrorObject(
+    error,
+    destType, // Preference is destination definition name
+    TRANSFORMER_METRIC.TRANSFORMER_STAGE.TRANSFORM
+  );
+  return getErrorRespEvents(
+    [input.metadata],
+    // eslint-disable-next-line no-nested-ternary
+    error.response
+      ? error.response.status
+      : error.code
+      ? error.code
+      : error.status || 400,
+    error.message || "Error occurred while processing payload.",
+    errObj.statTags
```

Comment:
We don't seem to be using anything other than `statTags` from `errObj`. 
How about moving the error message and status handling also to the `generateErrorObject` method?

Reply:
Actually it does handle message, status as well but since the router tf destination had already a kind of common implementation. We stuck to it!

Reply:
Yeah, but the handling is slightly different than this.
Any chance we can fix it now that we are refactoring this?

Reply:
I have changed some details, can you check and let me know if it confirms to how to you're imagining it ?